File: parent/pinyougou-interface/src/main/java/cn/itcast/core/service/SeckillOrderService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.seckill.SeckillOrder;
import entity.PageResult;

public interface SeckillOrderService {
    PageResult search (Integer page, Integer rows, SeckillOrder seckillOrder);

    void submitOrder (Long seckillId, String userId);

    SeckillOrder searchOrderFromRedis (String userId);

    void saveOrderFromRedis (String userId, Long orderId, String transactionId);

    void deleteOrderFromRedis (String userId, Long orderId);
}

File: parent/pinyougou-web-shop/src/main/webapp/js/service/seckillOrderService.js
app.service('seckillOrderService', function ($http) {
        this.findAll = function () {
            return $http.get('../seckillOrder/findAll.do');
        }
        //分页
        this.findPage = function (page, rows) {
            return $http.get('../seckillOrder/findPage.do?page=' + page + '&rows=' + rows);
        }
        //查询实体
        this.findOne = function (id) {
            return $http.get('../seckillOrder/findOne.do?id=' + id);
        }

        this.search = function (page, rows, searchEntity) {
            return $http.post('../seckillOrder/search.do?page=' + page + "&rows=" + rows, searchEntity);
        }
        //查询订单数
        this.findCount = function () {
            return $http.get('../seckillOrder/findCount.do');
        }
    }
)